helm/cluster-openstack/templates/_cluster.tpl

```diff
@@ -3,30 +3,33 @@ apiVersion: cluster.x-k8s.io/v1beta1
 kind: Cluster
 metadata:
   annotations:
-    cluster.giantswarm.io/description: "{{ .Values.clusterDescription }}"
+    cluster.giantswarm.io/description: {{ .Values.clusterDescription | quote }}
   labels:
+    {{- include "labels.common" . | nindent 4 }}
     cluster-apps-operator.giantswarm.io/watching: ""
-    {{- include "labels.common" $ | nindent 4 }}
-  name: {{ include "resource.default.name" $ }}
+  name: {{ include "resource.default.name" . }}
   namespace: {{ .Release.Namespace }}
 spec:
   topology:
-    class: {{ include "resource.default.name" $ }}
+    class: {{ include "resource.clusterClass.name" . }}
     version: {{ .Values.kubernetesVersion }}
     controlPlane:
       replicas: {{ .Values.controlPlane.replicas }}
       metadata:
         labels:
-          {{- include "labels.common" $ | nindent 10 }}
+          {{- include "labels.common" . | nindent 10 }}
     workers:
       machineDeployments:
+      {{- $outerScope := . }}
       {{- range .Values.nodePools }}
-      - class: {{ .class }}
-        name: {{ .name }}
-        replicas: {{ .replicas }}
+      {{- $innerScope := merge (deepCopy .) (deepCopy $outerScope) }}
+      {{- $nodeClass := merge (dict "name" .class) (deepCopy (get $outerScope.Values.nodeClasses .class)) (deepCopy $outerScope) }}
```

Comment:
this is a bit complex to understand, what about adding a comment about why you do it?